app/assets/javascripts/discourse/components/discourse-topic.js.es6

```diff
@@ -193,25 +172,65 @@ export default Ember.Component.extend(AddArchetypeClass, Scrolling, {
       }
     }
 
+    // Since the user has scrolled, we need to check the scroll direction on mobile.
+    // We use throttle instead of debounce because we want the switch to occur
+    // at the start of the scroll. This feels a lot more snappy compared to waiting
+    // for the scroll to end if we debounce.
+    if (this.site.mobileView && this.hasScrolled) {
+      Ember.run.throttle(
+        this,
+        this._mobileScrollDirectionCheck,
+        offset,
+        MOBILE_SCROLL_DIRECTION_CHECK_THROTTLE
+      );
+    }
+
     // Trigger a scrolled event
     this.appEvents.trigger("topic:scrolled", offset);
   },
 
-  // determines scroll direction, triggers header topic info on mobile
-  // and ensures that the switch happens only once per scroll direction change
-  mobileScrollGaurd(offset) {
-    // user hasn't scrolled past topic title.
-    if (offset < this.dockAt) return;
-
-    delta = offset - lastScroll;
-    // 3px buffer so that the switch doesn't happen with tiny scrolls
-    if (delta > 3 && scrollDirection !== "down") {
-      scrollDirection = "down";
-      this.appEvents.trigger("header:show-topic", this.topic);
-    } else if (delta < -3 && scrollDirection !== "up") {
-      scrollDirection = "up";
-      this.appEvents.trigger("header:hide-topic");
+  _mobileScrollDirectionCheck(offset) {
+    // Difference between this scroll and the one before it.
+    const delta = parseInt(offset - this._mobileLastScroll, 10);
```

Comment:
I believe you use `parseInt` here for type casting a float to int. What you could do is use `Math.floor`.